javascript/features/killtime/killtime_manager.js

```diff
@@ -54,6 +64,39 @@ class KilltimeManager {
         }
     }
 
+    //Gives all currently online players the weapon
+    giveAllPlayersKillTimeWeapon() {
+        if(this.weapon_ === null || this.weapon_ === undefined)
+            return;
+
+        server.playerManager.forEach(player => this.givePlayerWeapon(player));
+    }
+
+    // Give player entity a weapon if he's supposed to.
+    givePlayerWeapon(player) {
+        if (player.isNonPlayerCharacter())
+            return;
+        if(player.activity !== Player.PLAYER_ACTIVITY_NONE)
+            return;
+
+        player.giveWeapon(this.weapon_, 9999);
+    }
+
+    onPlayerSpawn(event) {
+        if(!this.isRunning_)
+            return;
+        
+        const player = server.playerManager.getById(event.playerid);
+        if (!player)
+            return;  // invalid |player| given for the |event|
+
+            
+        if(this.weapon_ === null || this.weapon_ === undefined)
```

Comment:
You can afaik also use !this.weapon_ for all falsy values.

Reply:
Falsyness wouldn't have passed here. (And personally I hate the idea of falsynes.. 
0==false === true. And 0 would theoratically be a valid number.

However I can easily validate against the actual supported values and will do so.